File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int el = Integer.MIN_VALUE;

        int ct = 0;

        for(int num : nums) {
            if (ct == 0) {
                el = num;
                ct++;
                continue;
            }
            if (num == el) {
                ct++;
            } else {
                ct--;
            }
            
        }

        return el;
    }
}

File: 0169-majority-element/0169-majority-element.go
func majorityElement(nums []int) int {
    var el int

    ct:=0

    for _, num := range nums {
        if ct == 0 {
            el = num
            ct++
            continue
        }
        if num == el {
            ct++
        } else {
            ct--
        }
        
    }

    return el
}